File: Gestor Hospitalario/Models/Usuario.cs
namespace Gestor_Hospitalario.Models
{
    public class Usuario
    {
        public int UsuarioID { get; set; }
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public SexoEnum Sexo { get; set; }
        public DateTime FechaNacimiento { get; set; }
        public string Direccion { get; set; }
        public string Telefono { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public RolEnum Rol { get; set; }

    }
    // Enumeración que define los valores válidos para el campo 'Sexo' en la base de datos.

    public enum SexoEnum
    {
        Masculino,
        Femenino
    }
    public enum RolEnum
    {
        Administrador,
        Medico,
        Paciente
    }
}

File: gestor-hospitalario-react/src/@types/usuario.ts
export interface Usuario {
  usuarioID: number;
  nombre: string;
  apellido: string;
  email: string;
  telefono: string;
  rol: 'Administrador' | 'Medico' | 'Empleado' | 'Paciente';
  fechaRegistro: string;
  activo: boolean;
}

export interface UsuarioLogin {
  email: string;
  password: string;
}

export interface UsuarioRegister {
  nombre: string;
  apellido: string;
  email: string;
  password: string;
  telefono: string;
  rol: 'Administrador' | 'Medico' | 'Empleado' | 'Paciente';
}

export interface UsuarioCreate extends UsuarioRegister {}

export interface UsuarioUpdate {
  usuarioID: number;
  nombre: string;
  apellido: string;
  email: string;
  telefono: string;
  rol: 'Administrador' | 'Medico' | 'Empleado' | 'Paciente';
  activo: boolean;
}

export type UserRole = 'Administrador' | 'Medico' | 'Empleado' | 'Paciente';

export interface RolePermissions {
  canRead: boolean;
  canCreate: boolean;
  canUpdate: boolean;
  canDelete: boolean;
}
